test/controllers/search_controller_test.rb

```diff
@@ -70,15 +70,16 @@ def test_pattern_search
 
     params = { search: { pattern: "", type: :google } }
     get(:pattern_search, params)
-    assert_redirected_to(controller: :rss_logs, action: :list_rss_logs)
+    assert_redirected_to(observations_path)
 
     params = { search: { pattern: "x", type: :nonexistent_type } }
     get(:pattern_search, params)
-    assert_redirected_to(controller: :rss_logs, action: :list_rss_logs)
+    assert_redirected_to(:root)
 
+    login("rolf")
     params = { search: { pattern: "", type: :observation } }
     get(:pattern_search, params)
-    assert_redirected_to(controller: :observations, action: :list_observations)
+    assert_redirected_to(observations_path)
```

Comment:
I added this last test to illustrate a bug.
The controller tries to redirect to `/observations/index`, see SearchController, ll. 46-48:
```ruby
      redirect_to(
        controller: ctrlr,
        action: :index
```
But the test says that the redirect is to `/`:
```ruby
$ rails t test/controllers/search_controller_test.rb
Started with run options --seed 59011

 FAIL["test_pattern_search", #<Minitest::Reporters::Suite:0x0000555db916e718 @name="SearchControllerTest">, 2.1695715410023695]
 test_pattern_search#SearchControllerTest (2.17s)
        Expected response to be a redirect to <http://test.host/observations> but was a redirect to <http://test.host/>.
        Expected "http://test.host/observations" to be === "http://test.host/".
        test/controllers/search_controller_test.rb:73:in `test_pattern_search'

  2/2: [========================================================================] 100% Time: 00:00:02, Time: 00:00:02

Finished in 2.17259s
2 tests, 28 assertions, 1 failures, 0 errors, 0 skips
```
Does this have something to do with the dual-root solution?
It also appears to me that `:root` may mean something different in the test than in the controller.

Reply:
Yes, good catch. I don't know offhand. This needs some investigation into the interaction of `:root` with path helpers, and as you said, possible differences between test/controller.

I'll start researching too.